lib/ramble/ramble/application.py

```diff
@@ -1419,9 +1443,168 @@ def format_context(context_match, context_format):
+                first_repeat_exp = repeat_exp_namespace
+
+        # Create initial results dict since analysis pipeline was skipped for base exp
+        self.results = {'name': base_exp_namespace}
+        self.results['N_REPEATS'] = n_repeats
+        self.results['EXPERIMENT_CHAIN'] = self.chain_order.copy()
+
+        #
+        # If repeat_success_strict is true, one failed experiment will fail the whole set
+        # and statistics will not be calculated
+        # If repeat_success_strict is false, statistics will be calculated for all successful
+        # experiments
+        #
+        repeat_success = False
+        exp_success = []
+        for exp in repeat_experiments.keys():
+            if exp in self.experiment_set.experiments.keys():
+                exp_inst = self.experiment_set.experiments[exp]
+            elif exp in self.experiment_set.chained_experiments.keys():
+                exp_inst = self.experiment_set.chained_experiments[exp]
+            else:
+                continue
+
+            exp_success.append(exp_inst.read_status())
+
+        if workspace.repeat_success_strict:
+            if 'FAILED' in exp_success:
+                repeat_success = False
+            else:
+                repeat_success = True
+        else:
+            if 'SUCCESS' in exp_success:
+                repeat_success = True
+            else:
+                repeat_success = False
+
+        if repeat_success:
+            self.set_status(status=experiment_status.SUCCESS)
+        else:
+            self.set_status(status=experiment_status.FAILED)
+
+        self.results['RAMBLE_STATUS'] = self.get_status()
+
+        if repeat_success or workspace.always_print_foms:
+            logger.debug(f'Calculating statistics for {n_repeats} repeats of {base_exp_name}')
+            self.results['RAMBLE_VARIABLES'] = {}
+            self.results['RAMBLE_RAW_VARIABLES'] = {}
+            for var, val in self.variables.items():
+                self.results['RAMBLE_RAW_VARIABLES'][var] = val
+                self.results['RAMBLE_VARIABLES'][var] = self.expander.expand_var(val)
+            self.results['CONTEXTS'] = []
+
+            results = []
+
+            # Iterate through repeat experiment instances, extract foms, and aggregate them
+            for exp in repeat_experiments.keys():
+                if exp in self.experiment_set.experiments.keys():
+                    exp_inst = self.experiment_set.experiments[exp]
+                elif exp in self.experiment_set.chained_experiments.keys():
+                    exp_inst = self.experiment_set.chained_experiments[exp]
+                else:
+                    continue
+
+                # When strict success is off for repeats (loose success), skip failed exps
+                if (not workspace.repeat_success_strict
+                    and exp_inst.results['RAMBLE_STATUS'] == 'FAILED'):
+                    continue
+
+                for context in exp_inst.results['CONTEXTS']:
```

Comment:
If a repeat experiment is missing the expected context (or if it doesn't have any results at all) this errors.

It should probably be guarded with something like `if 'CONTEXT' in exp_inst.results:` to make sure it has results to try and compute stats with.

And maybe the entire contents of this for loop could be guarded by this to prevent trying to iterate over results when they don't exist.